File: Java/exercicio2.java
class Carro {
    String marca;
    String modelo;
    int ano;
    int velocidade;

    Carro(String marca, String modelo, int ano) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.velocidade = 0;
    }

    void acelerar(int incremento) {
        this.velocidade += incremento;
    }

    void frear(int decremento) {
        this.velocidade -= decremento;
    }

    void exibirVelocidade() {
        System.out.println("Velocidade atual: " + this.velocidade + " km/h");
    }
}

public class Main {
    public static void main(String[] args) {
        Carro carro = new Carro("Toyota", "Corolla", 2020);
        carro.acelerar(50);
        carro.exibirVelocidade();
        carro.frear(20);
        carro.exibirVelocidade();
    }
}

File: Go/exercicio2.go
package main

import "fmt"

type Carro struct {
    Marca     string
    Modelo    string
    Ano       int
    Velocidade int
}

func (c *Carro) Acelerar(incremento int) {
    c.Velocidade += incremento
}

func (c *Carro) Frear(decremento int) {
    c.Velocidade -= decremento
}

func (c Carro) ExibirVelocidade() {
    fmt.Printf("Velocidade atual: %d km/h\n", c.Velocidade)
}

func main() {
    carro := Carro{"Toyota", "Corolla", 2020, 0}
    carro.Acelerar(50)
    carro.ExibirVelocidade()
    carro.Frear(20)
    carro.ExibirVelocidade()
}
